app/desmos/cmd/create-chain-link.go

```diff
@@ -0,0 +1,94 @@
+package cmd
+
+import (
+	"encoding/hex"
+	"io/ioutil"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/crypto/hd"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cobra"
+
+	"github.com/desmos-labs/desmos/app"
+	"github.com/desmos-labs/desmos/app/desmos/types"
+
+	profilescliutils "github.com/desmos-labs/desmos/x/profiles/client/utils"
+	profilestypes "github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+// GetCreateChainlinkFileCmd returns the command allowing to generate the chain link json file for creating chain link
+func GetCreateChainlinkFileCmd(generator types.ChainLinkReferenceGetter) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create-chain-link-file",
+		Short: "Generate the chain link json for creating chain link with the key",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			mnemonic, chain, err := generator.GetReference()
+			if err != nil {
+				return err
+			}
+
+			chainLinkJSON, err := generateChainLink(mnemonic, chain)
+			if err != nil {
+				return err
+			}
+
+			cdc, _ := app.MakeCodecs()
+			bz, err := cdc.MarshalJSON(&chainLinkJSON)
+			if err != nil {
+				return err
+			}
+
+			filename, _ := cmd.Flags().GetString("filename")
+			if strings.TrimSpace(filename) != "" {
+				if err := ioutil.WriteFile("data.json", bz, 0600); err != nil {
+					return err
+				}
+			}
+			return clientCtx.PrintBytes(bz)
+		},
+	}
+	flags.AddTxFlagsToCmd(cmd)
+	cmd.Flags().String("filename", "data.json", "The name of output chain link json file. It does not generate the file if it is empty.")
+	return cmd
+}
+
+// generateChainLink returns ChainLinkJSON for creating chain link
+func generateChainLink(mnemonic string, chain types.ChainType) (profilescliutils.ChainLinkJSON, error) {
+	// generate keybase for signing
+	keyBase := keyring.NewInMemory()
+	algo := hd.Secp256k1
+	hdPath := chain.DerivationPath
+	keyName := "chainlink"
+	_, err := keyBase.NewAccount("chainlink", mnemonic, "", hdPath, algo)
+	if err != nil {
+		return profilescliutils.ChainLinkJSON{}, err
+	}
+
+	// create the proof with the key
+	key, _ := keyBase.Key(keyName)
+	addr, _ := sdk.Bech32ifyAddressBytes(chain.Prefix, key.GetAddress())
+	sig, pubkey, err := keyBase.Sign(keyName, []byte(addr))
+	if err != nil {
+		return profilescliutils.ChainLinkJSON{}, err
+	}
+
+	// create chain link json
+	cdc, _ := app.MakeCodecs()
+	chainLinkJSON := profilescliutils.NewChainLinkJSON(
+		profilestypes.NewBech32Address(addr, chain.Prefix),
+		profilestypes.NewProof(pubkey, hex.EncodeToString(sig), addr),
+		profilestypes.NewChainConfig(chain.Name),
+	)
+	if err := chainLinkJSON.UnpackInterfaces(cdc); err != nil {
```

Comment:
What is this used for?

Reply:
It was used for json.Marshal. But, It has been replaced with cdc.Marshal already, I will remove it.